### PR TITLE
split apply combine of netcdf downloads

### PR DESCRIPTION
**Pull request by jread-usgs**

### CODE REVIEW

**Comment by jread-usgs** [comment]
Includes a progress bar for the split-apply-combine targets:
image

File: 6_drivers/src/nccopy_utils.R [hunk 1]
```diff
@@ -1,26 +1,125 @@
...
+  # need NCO to run this. Combine all split files into one:
+  system(sprintf('ncrcat -h --overwrite %s %s', filepaths_to_combine, nc_unl_filepath))
+
+
+  if (file.exists(nc_out_filepath)){
+    unlink(nc_out_filepath) # doesn't seem to honor the overwrite arg and gets hung up on interactive mode
+  }
+
+  # then make the time dim fixed again on the larger file
+  system(sprintf("ncks --fix_rec_dmn time %s -o %s", nc_unl_filepath, nc_out_filepath))
```
Comment (wdwatkins) [inline 6_drivers/src/nccopy_utils.R]: Not sure why the overwrite flag wouldn't work.  What were you trying?

Comment (jread-usgs) [reply 6_drivers/src/nccopy_utils.R]: I tried `--overwrite` and some other ones specified in the docs. Almost seemed like a bug or something...

Comment (wdwatkins) [reply 6_drivers/src/nccopy_utils.R]: hmm, maybe there was an issue with the arg order?  It happens in the terminal too, not just from `system()` in R?

Comment (wdwatkins) [reply 6_drivers/src/nccopy_utils.R]: Could report it on sourceforge then: https://sourceforge.net/p/nco/discussion/9829.

`nco` is up to date?

Comment (jread-usgs) [reply 6_drivers/src/nccopy_utils.R]: I don't think I did an exhaustive test before moving on w/ the simple `unlink` fix.

File: 6_drivers_prep.yml [hunk 2]
```diff
@@ -4,63 +4,74 @@ packages:
...
 
 sources:
   - 6_drivers/src/nldas_cube_utils.R
   - 6_drivers/src/nldas_feather_utils.R
-  
+
 targets:
-  6_drivers_prep: 
-    depends: 
+  6_drivers_prep:
```
Comment (aappling-usgs) [inline 6_drivers_prep.yml]: Is _prep intended to mean fetch+munge? I thought initially you had it meaning "prepare to fetch", but I'm pretty sure this file does a lot more than that now, right?

Comment (jread-usgs) [reply 6_drivers_prep.yml]: no, the names are off again. 6 now includes all of the fetch, so I need to rename before I start a major data pull

File: 6_drivers/src/nccopy_utils.R [hunk 3]
```diff
@@ -1,26 +1,126 @@
...
+  if (retry < retries){
+    invisible(file_out)
+  } else {
+    stop('\nexceded retries for nccopy', call. = FALSE)
   }
-  # check/verify file has the appropriate number of timesteps...fail if not
+}
+
+#' check that the .nc file output has the right number of timesteps
+verify_nc_output <- function(nc_filepath){
```
[the remark this answers is not in the document]
Comment (jread-usgs) [reply 6_drivers/src/nccopy_utils.R]: yes, great idea

**Pull request merged**